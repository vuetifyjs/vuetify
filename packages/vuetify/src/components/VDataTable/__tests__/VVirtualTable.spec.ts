// @ts-nocheck
/* eslint-disable */

// import VVirtualTable from '../VVirtualTable'
import {
  mount,
  Wrapper,
  MountOptions,
} from '@vue/test-utils'
// import Vue from 'vue'

describe.skip('VVirtualTable.ts', () => {
  type Instance = InstanceType<typeof VVirtualTable>
  let mountFunction: (options?: MountOptions<Instance>) => Wrapper<Instance>
  beforeEach(() => {
    mountFunction = (options?: MountOptions<Instance>) => {
      return mount(VVirtualTable, options)
    }
  })

  it('should render', () => {
    const vm = new Vue()

    const wrapper = mountFunction({
      propsData: {
        items: ['a', 'b', 'c'],
      },
      scopedSlots: {
        items: props => vm.$createElement('div', { staticClass: 'test' }, [JSON.stringify(props)]),
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should re-render when items change', async () => {
    const wrapper = mountFunction({
      propsData: {
        items: ['a', 'b', 'c'],
      },
      scopedSlots: {
        items (props) {
          return this.$createElement('div', props.items.map(i => this.$createElement('div', [i])))
        },
      },
    })

    expect(wrapper.html()).toMatchSnapshot()

    wrapper.setProps({
      items: ['d', 'e', 'f'],
    })

    expect(wrapper.html()).toMatchSnapshot()
  })
})
