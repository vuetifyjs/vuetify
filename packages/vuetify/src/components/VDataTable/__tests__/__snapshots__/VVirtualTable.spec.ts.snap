// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VVirtualTable.ts should re-render when items change 1`] = `
<div class="v-data-table v-virtual-table theme--light">
  <div class="v-virtual-table__wrapper">
    <div class="v-virtual-table__table">
      <table>
        <tbody>
          <tr style="height: 0px;">
          </tr>
          <div>
            <div>
              a
            </div>
            <div>
              b
            </div>
            <div>
              c
            </div>
          </div>
          <tr style="height: 0px;">
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;

exports[`VVirtualTable.ts should re-render when items change 2`] = `
<div class="v-data-table v-virtual-table theme--light">
  <div class="v-virtual-table__wrapper">
    <div class="v-virtual-table__table">
      <table>
        <tbody>
          <tr style="height: 0px;">
          </tr>
          <div>
            <div>
              d
            </div>
            <div>
              e
            </div>
            <div>
              f
            </div>
          </div>
          <tr style="height: 0px;">
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;

exports[`VVirtualTable.ts should render 1`] = `
<div class="v-data-table v-virtual-table theme--light">
  <div class="v-virtual-table__wrapper">
    <div class="v-virtual-table__table">
      <table>
        <tbody>
          <tr style="height: 0px;">
          </tr>
          <div class="test">
            {"items":["a","b","c"]}
          </div>
          <tr style="height: 0px;">
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;
